icu4c/source/i18n/usearch.cpp

```diff
@@ -1432,10 +1432,11 @@ const CEI *CEIBuffer::get(int32_t index) {
     //   Verify that it is the next one in sequence, which is all
     //   that is allowed.
     if (index != limitIx) {
-        U_ASSERT(FALSE);
-        // TODO: In ICU 64 the above assert was changed to use UPRV_UNREACHABLE instead
-        // which unconditionally calls abort(). However, there were cases where this was
-        // being hit. This change is reverted for now, restoring the existing behavior.
+        UPRV_UNREACHABLE_ASSERT;
+        // TODO: In ICU 64 the above was changed from U_ASSERT to UPRV_UNREACHABLE,
+        // which unconditionally called abort(). However, there were cases i which
+        // it was being hit, so it was changed back to U_ASSERT in ICU-20680. Now
```

Comment:
Just a minor suggestion: Maybe instead of "Now", we could use the version number (similar to the above ICU 64)?
```suggestion
        // it was being hit, so it was changed back to U_ASSERT in ICU-20680. In ICU 70, we now use the new UPRV_UNREACHABLE_ASSERT
```